### PR TITLE
Allow a clean shutdown on window system failure

### PR DESCRIPTION
**Pull request by lrusak**
I have been investigating why we segfault when windowing cannot be initialized.

This allows a clean shutdown if `false` is return somewhere in `InitWinSystem()`

If this isn't desired it would still be nice to add the nullptr checks

### CODE REVIEW

**Comment by FernetMenta** [comment]
nice. I misinterpreted the intent of this code in the morning. This looks good

**Review by FernetMenta — APPROVED** [review]

**Comment by lrusak** [comment]
@FernetMenta thank you. Please note that the GLES renderer changes were do to some odd behaviour in the way `std::unique_ptr<CGLESShader*[]> m_pShader;` was working with `nullptr` checks.

It seems that 
```
if (m_pShader[SM_DEFAULT])
```
was passing the nullptr check for some reason and this would cause a segfault in the `Free()`

I'm not sure why this was or if someone can explain this behaviour to me. Regardless I think the `std::array<std::unique_ptr<CGLESShader>, SM_MAX> m_pShader;` format looks better anyways.

If you want me to do this to GL just say so unless you want to do it 😉

**Comment by lrusak** [comment]
@FernetMenta care to give this a test before merging?

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 1]
```diff
@@ -680,38 +672,24 @@ void CRenderSystemGL::ReleaseShaders()
 {
   if (m_pShader[SM_DEFAULT])
     m_pShader[SM_DEFAULT]->Free();
-  delete m_pShader[SM_DEFAULT];
```
Comment (FernetMenta) [inline xbmc/rendering/gl/RenderSystemGL.cpp]: any reason why you don't release the smart pointer here? you can't call m_pShader[SM_DEFAULT]->Free() two times in a row but you allow this with this code.

Comment (lrusak) [reply xbmc/rendering/gl/RenderSystemGL.cpp]: Yes that's a good point, let me fix that.

**Comment by Rechi** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
I tested the GL commit here: works

**Pull request merged**